src/main/java/com/neroreflex/pizza/Pizza.java

```diff
@@ -142,76 +142,52 @@ protected void onMessage(String channel, String sender, String login, String hos
             // Controlla se il trancio e' presente e registrato
             if (this.tranci.containsKey(command)) {
                 // E se lo e' piazza la richiesta che dovra' prendere in carico
-                if (this.tranci.get(command).isLoaded())
-                    this.tranci.get(command).enqueueRequest(new Request(channel, sender, rawParamsString));
-                else
-                    this.sendMessage(channel, sender + " sorry but the requested plugin is not loaded (yet)");
-            } else if (command.compareTo("help") == 0) {// E se non lo e' invia l'errore
-                this.Help(new Request(channel, sender, rawParamsString));
-            } else if (command.compareTo("info") == 0) {// E se non lo e' invia l'errore
-                this.Info(new Request(channel, sender, rawParamsString));
+                if (this.tranci.get(command).isLoaded()) {
+                    this.tranci.get(command).enqueueEvent(new Event(EventType.UserRequest, channel, sender, rawParamsString));
+                }
+                else {
+                    this.sendMessage(channel, sender + " unavailable plugin.");
+                }
+            } else if (command.equals("help")) {
+                this.Help(new Event(EventType.HelpRequest, channel, sender, rawParamsString));
+            } else if (command.equals("info")) {
+                this.enqueueMessage(new Message(channel, "I'm PizzaBot (https://github.com/NeroReflex/Pizza) a small and modular IRC bot"));
             } else {
                 this.enqueueMessage(new Message(channel, "The request '" + command + "' cannot be resolved"));
             }
-        } else if (message.substring(0, nickLength).compareTo(this.getNick()) == 0) {
+        } else if (message.substring(0, nickLength).equals(this.getNick())) {
             this.enqueueMessage(new Message(channel, "type !help to get started"));
         }
     }
 
     /**
-     * Il bot si presenta alla richiesta di informazioni
-     *
-     * @param helpRequest la richiesta di informazioni
-     */
-    public final void Info(Request helpRequest) {
-        this.enqueueMessage(new Message(helpRequest.getChannel(), "I'm PizzaBot (https://github.com/NeroReflex/Pizza) a small and modular IRC bot"));
-    }
-
-    /**
      * Il bot alla richiesta di aiuto fornisce la lista dei plugin e il loro
      * utilizzo.
      * Si puo' anche richiedere aiuto in uno specifico plugin.
      *
      * @param helpRequest la richiesta di aiuto
      */
-    public final void Help(Request helpRequest) {
+    public final void Help(Event helpRequest) {
         // Dividi la richiesta di aiuto in piu' parti
-        Vector<String> args = helpRequest.getBasicParse();
+        Vector<String> args = new Vector<>(Arrays.asList(helpRequest.getInfo().get(2).split("\\s+"))); 
+        args.removeIf(i -> i.equals(""));
```

Comment:
i -> i.isEmpty();
Alla fine si è capito come ci finiscono le stringhe vuote? accade solo nel caso in cui la stringa sia vuota?